script/verification/HoloceneSystemConfigUpgrade.s.sol

```diff
@@ -0,0 +1,279 @@
+// It also verified that remaining storage variables are unchanged.
+contract HoloceneSystemConfigUpgrade is SuperchainRegistry {
+    using LibString for string;
+
+    address public systemConfigAddress;
+    ISystemConfig sysCfg;
+
+    // Storage variables which exist in the existing and target
+    // SystemConfig contract versions irrespective of upgrade path,
+    // and which should be preserved:
+    struct BaseSysCfgVars {
+        address owner;
+        bytes32 batcherHash;
+        uint256 gasLimit;
+        address unsafeBlockSigner;
+        IResourceMetering.ResourceConfig resourceConfig;
+        address batchInbox;
+        address l1CrossDomainMessenger;
+        address l1StandardBridge;
+        address l1ERC721Bridge;
+        address optimismPortal;
+        address optimismMintableERC20Factory;
+    }
+
+    BaseSysCfgVars previous;
+
+    // Values which may change during the upgrade, depending on the chain::
+    uint256 previousScalar;
+
+    // The target value for the DisputeGameFactory address,
+    // this getter does not exist in the SystemConfig contract @ 1.12.0:
+    address targetDGF;
+
+    // Target values for gasPayingToken and its decimals. This should represent ETH
+    // on all upgrade paths:
+    address constant targetGasPayingToken = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+    uint8 constant targetDecimals = 18;
+
+    // The target version of the SystemConfig contract, read from the SCR @ specified release:
+    string targetVersion;
+
+    // The constructor caches some information from the Superchain Registry and from the existing SystemConfig contract
+    // before the upgrade is executed.
+    constructor(string memory _l1ChainName, string memory _l2ChainName, string memory _release)
+        SuperchainRegistry(_l1ChainName, _l2ChainName, _release)
+    {
+        systemConfigAddress = proxies.SystemConfig;
+        sysCfg = ISystemConfig(proxies.SystemConfig);
+
+        // cache the values of the SystemConfig contract before the upgrade
+        previous = getBaseSysCfgVars();
+        previousScalar = sysCfg.scalar();
+
+        // Read target version from SCR @ specified release
+        targetVersion = standardVersions.SystemConfig.version;
+
+        if (sysCfg.version().eq("2.3.0")) {
+            // Supported initial version
+            targetDGF = sysCfg.disputeGameFactory();
+        } else if (sysCfg.version().eq("2.2.0")) {
+            // Supported initial version
+            targetDGF = sysCfg.disputeGameFactory();
+        } else if (sysCfg.version().eq("1.12.0")) {
+            // Supported initial version
+            targetDGF = address(0);
+        } else {
+            revert("unsupported SystemConfig version");
+        }
+    }
+
+    /// @notice Public function that must be called by the verification script.
+    function checkSystemConfigUpgrade() public view {
```

Comment:
Do you check for parity of the `resourceConfig`, or is this not necessary? If not, and you can't import the `ISystemConfig` interface from the monorepo, you could trim down the interface you define below in the file to only those functions you need, and also remove the `IResourceMetering` and its contained struct.

Reply:
I do check for parity for `resourceConfig`.

Reply:
sorry yes, by hashing the whole thing 👑